packages/react-components/lib/tasks/utils.ts

```diff
@@ -20,6 +20,32 @@ export function taskTypeToStr(taskType: number): string {
   }
 }
 
+export interface TaskRequestLabel {
+  category?: string;
+  unix_millis_warn_time?: number;
+  pickup?: string;
+  destination?: string;
+  cart_id?: string;
+}
+
+export function parseTaskRequestLabel(taskState: TaskState): TaskRequestLabel | null {
+  let requestLabel: TaskRequestLabel | null = null;
+  if (taskState.booking.labels) {
+    for (const label of taskState.booking.labels) {
+      try {
+        const parsedLabel: TaskRequestLabel = JSON.parse(label);
```

Comment:
This does not validate the json, technically the correct return value is `unknown`. We should at least check that

1. it is an object
2. all of the fields have the correct type